src/client.rs

```diff
@@ -111,28 +138,124 @@ mod test {
         let child = thread::spawn(move || {
             let (mut connection, _)  = test_server.accept().unwrap();
 
+            // First proposal should be fine, no errors.
+            scoped_debug!("Should get preamble and proposal. Responds Success");
+
             // Expect Preamble.
             let message = serialize::read_message(&mut connection, ReaderOptions::new()).unwrap();
             let preamble = message.get_root::<connection_preamble::Reader>().unwrap();
             // Test to make sure preamble has the right id.
             if let connection_preamble::id::Which::Client(Ok(id)) = preamble.get_id().which().unwrap() {
+                scoped_debug!("got preamble");
                 assert_eq!(Uuid::from_bytes(id).unwrap(), client_id);
             } else { panic!("Invalid preamble."); }
 
-            // Expect proposal.
+            // Expect first proposal! (success!)
             let message = serialize::read_message(&mut connection, ReaderOptions::new()).unwrap();
             let request = message.get_root::<client_request::Reader>().unwrap();
             // Test to make sure request has the right value.
             if let client_request::Which::Proposal(Ok(proposal)) = request.which().unwrap() {
+                scoped_debug!("got proposal");
                 assert_eq!(proposal.get_entry().unwrap(), to_propose);
             } else { panic!("Invalid request."); }
 
-            // Send a response.
+            // Send first response! (success!)
             let response = messages::proposal_response_success();
             serialize::write_message(&mut connection, &*response).unwrap();
+            connection.flush();
+
+            // Second proposal should report unknown leader, and have the client return error.
+            scoped_debug!("Should get proposal. Responds UnknownLeader");
```

Comment:
One thing I've learned with Rust, though, is that line numbers on panics only come from the site of the panic, so it's best to not make assertions in test helper functions, and instead make the assertions on the result of calling them.